modulus/models/fcn_mip_plugin.py

```diff
@@ -159,21 +159,25 @@ def __init__(
     def prepare_input(self, input, time):
         device = input.device
         dtype = input.dtype
-        num_chans = input.size(2)
-        input_list = list(torch.split(input, 1, dim=1))
+
         i = self.input_map_wts.row.values - 1
         j = self.input_map_wts.col.values - 1
         data = self.input_map_wts.S.values
-        M = torch.sparse_coo_tensor(np.array((i, j)), data).to(device).type(dtype)
-
-        for i in range(len(input_list)):
-            input_list[i] = input_list[i].reshape(num_chans, -1) @ M.T
-            input_list[i] = input_list[i].reshape(1, num_chans, 6, 64, 64)
+        M = torch.sparse_coo_tensor(np.array((i, j)), data).type(dtype).to(device)
 
+        input = input.reshape(input.shape[0], input.shape[1], input.shape[2], -1) @ M.T
+        input = input.reshape(input.shape[0], input.shape[1], input.shape[2], 6, 64, 64)
```

Comment:
Can we abstract the size of the underlying grid?

Reply:
Great comment, done!